aten/src/ATen/native/cuda/Embedding.cu

```diff
@@ -320,7 +316,7 @@ Tensor & embedding_renorm_cuda_(Tensor & self, const Tensor & indices,
 
     AT_DISPATCH_FLOATING_TYPES_AND2(at::ScalarType::Half, at::ScalarType::BFloat16, self.scalar_type(), "embedding_backward", [&] {
       using accscalar_t = acc_type<scalar_t, true>;
-      renorm_kernel<<<grid, block, 128 * sizeof(accscalar_t), stream>>>(
+      renorm_kernel<<<grid, block, (block.x / C10_WARP_SIZE) * sizeof(accscalar_t), stream>>>(
```

Comment:
Block size here is a constant 128 so is always a multiple of `C10_WARP_SIZE`.

Reply:
Yep, to be on the safe side, can you please contexpr `block` and put in a `static_assert` to verify? Remember, some builds require 2 arg version of static_assert, I'm constantly bitten by this.